### DISCUSSION TITLE
‼️`Venus`组件`v1.7.0`使用注意事项

### DISCUSSION THREAD

**Discussion by simlecode** [filecoin-project/venus#5265] [Announcements]
## 各组件版本

组件 | 版本号
--- | ---
venus | v1.7.1
venus-auth | v1.7.0
venus-wallet | v1.7.0
venus-messager | v1.7.0
venus-gateway | v1.7.0
venus-miner | v1.7.0
venus-market | v2.3.0

### 调整 venus-gateway 验证 venus-wallet 持有地址是否合法的逻辑

涉及组件：`venus-gateway` `venus-wallet`

**非常重要：`venus-gateway` `v1.7.0` 版本需要兼容 `venus-wallet` `v1.7.0` 以下版本，`venus-gateway` 要先于 `venus-wallet` 升级到 `v1.7.0`**

**兼容方案**：`venus-gateway` 新增加一个flag `disable-address-verify`，用于表示是否验证 `venus-wallet` 注册上来的地址是否合法

* `./venus-gateway run --disable-address-verify` **不会验证钱包地址的合法性，当 `venus-wallet` 版本低于 `v1.7.0` 时，`venus-gateway` 启动需要加上 `--disable-address-verify`**
* `./venus-gateway run` **会验证钱包地址的合法性，`venus-wallet` 使用的版本需要大于或等于 `v1.7.0`**

`venus-wallet` 和 `venus-gateway` 升级后都需要检查 `venus-wallet` 的地址是否正常注册到 `venus-gateway`，
使用命令 `./venus-gateway wallet list` 可以列出所有注册到 `venus-gateway` 的地址。

### 调整本地 jwt token

涉及组件：`venus` `venus-messager` `venus-gateway`

具体修改：每次启动都会生成新的jwt token，且 `token` 都保存在 `~/.repo_path/token`

### 增加 metrics

设计组件：venus-miner venus-messager venus-gateway

具体使用请参考文档：

* [venus-miner](https://github.com/filecoin-project/venus-miner/blob/master/docs/zh/config-desc.md#metrics-%E9%85%8D%E7%BD%AE%E9%A1%B9%E8%A7%A3%E6%9E%90)
* [venus-messager](https://github.com/filecoin-project/venus-messager/blob/master/docs/zh/metrics-config-desc.md)
* [venus-gateway](https://github.com/ipfs-force-community/venus-gateway/blob/master/docs/zh/metrics-config.md)

### 增加 Version 接口

会返回当前程序的版本

涉及组件：venus-auth，venus-messager，venus-gateway，venus-market

具体使用参考：https://github.com/filecoin-project/venus/issues/5132#issue-1316366156

###  venus-gateway 新增repo目录
1. 增加 `repo` 目录，默认在 `~/.venusgateway`，并包含两个文件：`config.toml` 和 `token`，`token` 是本地的jwt token，给命令行使用
2. venus-gateway第一次启动时,会自动在`--repo` flag 指定的目录(默认为`~/.venusgateway`)根据启动参数自动生成配置文件, 如果后续需要修改配置, 可以直接通过修改配置文件来完成.

**注意**💁 :
 - 命令行flag的优先级高于配置文件的优先级. 命令行flag与配置文件中相应的项不一致时, 优先使用命令行的参数项.
 - 配置文件只有在第一次生成时会根据命令行参数来生成, 之后不会被重写.

### venus-messager

* 移除 `--config` flag

**Comment by diwufeiwen** [comment]
venus-market感觉在签名重构版本合进去前也应该发个版本

**Reply by simlecode** [reply]
会发 v2.3.0
